Title: Israel O. Dil√°n Pantojas
Date: 2016-01-22
Modified: 2017-02-07
Slug: israel
Category: People
Tags: de-novo

## Contact info:

 - e-mail - <israelodilan@gmail.com>
 - Github - <https://github.com/Omig12/>

# Bio(3):

## New-New Weekly update:
#### _iBRIC summer internship at University of Pittsburgh, it's gonna be a blast and I'll keep you posted._

------------------------------
![Causal Bayesian Networks!](http://www.abiomics.eu/media/images/oksagi.jpg)

<h4> Image taken from Google </h4>

-------------------------------

### Week 4: (12-16/Jun)
- Center for Causal Discovery short course
- Main topics Discussed:
  - Causal Networks
  - Bayesian Statistics
  - Bayesian Causal Networks
  - Tetrad Causal Discovery Software
- Datathon Thursday
- Journal Club Meeting on Friday

### Week 3: (5-9/Jun)
- During this week, I was instructed to do the following:
  - Get the Tetrad software up and running in my computer
  - Read up on and understand the [(Peter Spirtes and Clark Glymour) PC Algorithm](https://www.cs.cmu.edu/afs/cs.cmu.edu/project/learn-43/lib/photoz/.g/scottd/fullbook.pdf)  
  - Observational Data Representation within causal networks
  - Experimental Data Representation within causal networks
  - Ethics Forum Group presentation on Michael A. Ballesies Case _Arming America_ coming soon.
- **Research topic update**:
 - _Methods for integrating both Observational and Experimental Data in Constraint-Based Causal Discovery Algorithms_ for Causal Discovery in Causal Bayesian Networks.  

### Week 2: (30/May-2/Jun)
- During this week we have been asked to write a brief proposal due Friday 2nd.
  -  [Proposal](https://docs.google.com/document/d/1gj9t59rjuQFGE9DD3b8DEYTxMYyrs9QruzyrLgozMGE/edit?usp=sharing)
- Reading a lot of literature in order to understand Causal Bayesian Networks, d-separation and the Causal Markov condition.

### Week 1: (22-26/May)
- Was assigned to Dr. Gregory Cooper's Lab at the Pitt Department of Biomedical Informatics in the Center for Causal Discovery.
- Was required to complete ethics in research training modules.
  - Responsible Conduct of Research
  - Biomedical Informatics Research ethics
- **Research topic**:
  - Causal Discovery utilizing Causal Bayesian Networks Algorithms
- Started reading up on all the required statistics and graph theory required to work with the topic.
- Currently reading the first chapter of:
> [Glymour, C. N., & Cooper, G. F. (Eds.). (1999). Computation, causation, and discovery. Aaai Press.](http://fitelson.org/269/Cooper_Intro.pdf)

---------------------------------------------

# Bio(2):

 I have respawned :) Muahahahahhah! No but for real, I'm back, doing research at the Megaprobe-Lab in the hopes of expanding my knowledge of bioinformatics, and general programing.

## New Weekly UPDATE (Jan 2017 - May 2017)

### Week 11-19:
- Classes still interrupted.

### Week 10: (20-27/Mar)
- Campus went on strike protesting budget cuts.
- Classes suspended till further notice.

### Week 9: (13-17/Mar)
- Starting to write Python Makefile [Snakemake tutorial](http://slides.com/johanneskoester/snakemake-tutorial-2016#/18)
- [Snakemake documentation](https://snakemake.readthedocs.io/en/stable/snakefiles/writing_snakefiles.html)

### Week 8: (6-10/Mar)
- Professor suggested utilizing [SnakeMake](!https://snakemake.readthedocs.io/en/stable/) so far it's a really good prospect.
- Read [this SnakeMake](https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/bts480) paper for more info.

### Week 7: (27/Feb-3/Mar)
- Found a more pythonic way of finding information about device resources while reading up on the python standard library documentation.
- Found interesting alternative to make called [Do-it](http://pydoit.org/).

### Week 6: (20-24/Feb)
- Very anxious this Week.
- Student strike.
- Got sick :'(

### Week 5: (13-17/Feb)
- Did not work in the lab this week.
- Most time was spent studying.

### Week 4: (6-10/Feb)
- Pipeline device resource detection script:
  - Return available CPU cores
  - Return available RAM free

### Week 3: (1-5/Feb)
- Working on pipeline makefile.
- Learning about Trinity.

### Week 2: (25-29/Jan)
- Started work on pipeline makefile.
- Looking at Diginorm.

### Week 1: (19-22/Jan)
- Started reading up on the [khmer paper!](https://f1000research.com/articles/4-900/v1)
- Joined the Lab :') YEY!

------------------------------------------------------
# Bio:

I confess my crimes against humanity and Eleutherodactylus coqui races alike, for I have vanquished the furious lab frog.
> Now, I am become Death, the destroyer of worlds.
J.R. Oppenheimer

[![Seriously!](https://s-media-cache-ak0.pinimg.com/564x/2e/b4/63/2eb4631511b658d831ee851538d0673b.jpg)]

# Research Goals

Create and analyze correctly obtained De Bruijn graphs from de-novo sequence assemblies utilizing mutual.

## Weekly UPDATE (Jan 2016 - May 2016)

### Week 17: (9/May-13/May)
- Got down to working on the technical report.
- Cleaned and updated the repo:
  - Improved formatting on documents.

### Week 16: (2/May-6/May)
- Worked Hard on fixing parser issues.
- Got some output, lightly checked/skimmed output for errors, did not find any.
- Continued working on technical report.
- To do:
  - Finish technical report.

### Week 15: (25-29/Apr)
- Began corroboration of correct parsing of LastGraph file:
- Noticed wrong data being displayed in parser's GFA's output of Read Count.
- Noticed Problem in Link report reported from Bandage's output vs. Parsers output.
- Noticed Problem in reported Link's nodes orientation from Parser.
- Changed the way parser stored the data from first storing it in an array to now utilizing networkx's built-in structure for storing attribute.
- Fixed wrong read count report.
- Kinda theoretically fixed links being reported and node orientation.
- Parser is temporarily broken due to issues with networkx structure.
- To do:
  - Fix issues that are preventing the parser from executing correctly.
  - Continue technical report.

### Week 14: (18-22/Apr)
- Started working on LastGraph parser in python:
- Reads file
- Finds NODE
- Appends to list
- Finds ARC
- To done :)
  - Write all contigs obtained from De-Bruijn
  - Determine format for output
  - Presumably got a working parser now.
  - New to do:
  - Corroborate parser works
  - Keep improving it
  - Initiate technical report

### Week 13: (11-15/Apr)
- Hulk got back up (still has some fight left).
- Ran Mutual twice simultaneously:
  - A) With the purpose of comparing organism's A (input velvet/oases output [velvetA]) De-Bruijn graph with organism's B (input velvet/oases output [velvetB]) De-Bruijn graph after the data had been cleaned with sickle and scythe.
  - B) With the purpose of comparing organism's A with organism's A (self comparison) expecting duplicate output from Mutual (outputa.fa = outputb.fa).
- Created a database using Blast's formatdb on Mutual's output (outputa.fa) to compare other .fa files against it.
- Blasted through Mutual's output (outputa.fa) database against Velvet's ouput (contigs.fa)
- Blasted through Mutual's output (outputa.fa) database against Mutual's ouput (outputb.fa)
- To do:
  - Analyse results see if comparisons results were accurate.

### Week 12: (4-8/Apr)
- Many evaluations this week also :'(, worked mostly on preparing a bioinformatics presentation that aids in explaining Mutual's Heuristic Pairwise Alignment algorithm implementation.
- Was corrected by professor need to parse Velvelt's LastGraph file (that is to say it's De-Bruijn graph), not the fasta/fastq file.
- Good news everyone! Found a LastGraph parser in written Ruby. Bad news everyone! I don't understand ruby. :{
  - Tried to ran Mutual but The Hulk(server) is down.
  - To do:
  - Turn to Velvet/Oases De-Bruijn parser.
  - Need to run clean data on Hulk.

### Week 11: (28/Mar-1/Apr)
  - Utilized Bandage to create a few graphs of interesting contiguous nodes.
  - Experimented a few forms of output including GFA extensions to graph.
  - Started reading up on parsers for fasta/fastq files. Using Biostars <https://www.biostars.org/p/710/> Guides  and Nacrokill's <https://github.com/NacroKill/dmel-ercc-diff/blob/master/Tophat_Birnavirus_Confirmer_2.0.py> parsers for referrence.
  - Taking a bit of a break many exams this and the next week.

### Week 10: (21-25/Mar)
- Holy Week/Spring break.

### Week 9: (14-18/Mar)
 - Ran clean data through Velvet/Oases.
 - Graphed Output of Organism A.
    - Noticeable results on graph.
 - Preparing to run files on mutual.
 - Student assembly paralyzed work.

![Bandage graph of velvet/oases output.]({filename}images/NewContigs_graph.png)


### Week 8: (7-11/Mar)
 - Git repo is finally in order. :)
 - Finally got to quality trimming:
 - Needed Illumina Adapters FASTA files to remove adapters, found them in Trimmomatic source code. <http://www.usadellab.org/cms/?page=trimmomatic>
 - Helpful note utilize FastQC to help determine how to quality control your reads. Especially figuring out the encoding (In this case Illumina/Sanger Encoding).
 - Apparently scythe supports gunzipped fastq files.
 - Finally removed Adapters
 - Finally trimmed Edges
 - Created quality reports using FASTQC for each step.
 - To do:
   - Verify if quality can be further improved.
   - Re-run mutual with clean data and graph outputs
   - Close to source code examination.

### Week 7: (29/Feb-4/Mar)
 - Reinstalled Ubuntu. :'D (So shinny!)
 - Updated Weekly.
 - Many a exams this week.
 - Began reading "The Cartoon Guide To Genetics" by Larry Gonick & Mark Wheelis.
 - Analyze best way to compare outputs from velvet/oases and mutual.
 - Options:
   - Find a way to graph mutuals output compare graphs with velvet's.
   - Also possibly compare with output of other that does analysis between multiple organisms.
   - Utilize blast to compare each contig.
   - Utilize clustel to compare contig alignment.
  - Still to do:
    - Remove Adapters with Scythe.
    - Trim edges with Sickle.

### Week 6: (22-26/Feb)
- Fought with git and github's repos in an effort to sync files, failed.
- Switching strategy: Prioritizing some file to be copied and leaving other ones behind stored only locally in both hulk and personal machine.
- Installed Scythe and Sickle.
- Graphed Contigs from velvet's LastGraph utilizing Bandage.
- Local Computer's Ubuntu OS broke  (RIP) :'(  Thankfully to prior advice given in a lab meeting, had created multiple back ups of research data.
  - Read a few more papers concerning bioinformatics and its history.

  ![Bandage graph of velvet/oases output.]({filename}images/contigs_graph.png)
  ![Starlet Sea Anemone for comparisson.] (http://www.kahikai.org/images/embryology_development/nv.jpg)

### Week 5: (15-20/Feb)
- Worked on creating research documentation and a research log:
- Wrote FAQ to help ease the reproduction of this research.
- Experienced many issues syncing local and remote git repo.
- Verified Mutual's output using two different blast services both returned correctly identified nemastotella similarity.
- Verified Quality of input files using FastQC, file are very high quality reads.
- To do:
  - Visualize contigs using Bandage.
  - Trim Illumina adapters from input sequences.

### Week 4: (8-12/Feb)
- Setting up Hulk server to run mutual. (Still missing velvet)
- Battled with Hulk to run mutual.
- Led to a crash :'( after two days of work.
  - Finally got it down, ran mutual and got some output.
  - Initiated analyzing phase, to do:
  - Verify Mutual's output utilizing blast <http://blast.ncbi.nlm.nih.gov/Blast.cgi> and blastn <http://genome.jgi.doe.gov/pages/blast-query.jsf?db=Nemve1>
  - Clean up initial data by removing adapters using Sickle and Scythe <http://bioinformatics.ucdavis.edu/research-computing/software/>.
  - Visualize mutual outputs with the help of bandage <https://github.com/rrwick/Bandage>.
  - Started working on a more readable and reproducible version of the documentation for this research.

### Week 3: (1-5/Feb)

- Initiated test running of mutual on local (DELL Inspiron 7548) machine, utilizing sample Sea Anemone Dataset.
- Estimated run-time: 72 hour.
- Expected output: A transcript with very long contigs due to the high similarity of the two test organisms.
- Results: Two empty output files. >:'(
  - Possible issues: Computer might not meet requirements to performs such heavy work very efficiently.
  - Path of action: Proceeded to move operations elsewhere, The University of Puerto Rico's Computer Science Department very own Hulk.
  - Rest of the week(get it?) due to health problems.

### Week 2: (25-29/Jan)

- Installed required programs (velvet/oases) to test out mutual.
- Acquired Sample Data Set of Nemastotella Embryonic Transcriptome (Starlet sea anemone). <https://darchive.mblwhoilibrary.org/handle/1912/5613>
- Working on glossary of Bioinformatics related terms.
- Continued reading up on de-novo sequences, mutual/velvet/oases and De-Bruijn Graph assembly of short reads.
- Test ran mutual.


### Week 1: (19-22/Jan)

- Started reading up on de-novo sequencing and familiarizing with the terms used around it.
